src/Stratis.Bitcoin/P2P/PeerSelector.cs

```diff
@@ -194,12 +194,25 @@ private IEnumerable<PeerAddress> SelectPreferredPeers()
                 this.logger.LogTrace("[RETURN_ATTEMPTED_HC_FAILED]");
                 return attempted;
             }
+            
+            int attemptedReachedThresholdCount = this.peerAddresses.Values.Count(p => p.ConnectionAttempts == PeerAddress.AttemptThreshold && !this.IsBanned(p));
```

Comment:
Please add an extension method for `peerAddresses` so that you can go `this.peerAddresses.NotBanned()` and set it to a field in the method. t=That way we cache to result and only work with that result set for the scope of the method.

Reply:
Good idea, fixed